### ISSUE TITLE
Update the “Edit site” admin bar control to open the Site Editor at the top-level, consistently

### ISSUE THREAD

**Issue by richtabor** [WordPress/gutenberg#63785]
Currently the "Edit site" control in the Admin Bar brings you directly to the associated template you’re viewing, within the template editing view.  

Instead, I propose that this "Edit site" control should _always_ open the Site Editor at the top level, just like the "Editor" control does within the WP Admin > Appearance menu—rather than on the template view of whichever page you're on. 

### Why?

The current experiences is disorienting as-is, dropping you into various levels of the Site Editor. This behavior makes the control loose its effectiveness as an entry point to the Site Editor, making it unpredictable where you will actually land. 

For higher-level control like editing a specific template you're viewing on the front-end, I'd expect perhaps Create Block Theme could add a toolbar control to do so. 

Most people just want to edit their site, and starting from one consistent point is a clear way to reduce the heaviness of navigating the site editing experience. 

### Visual 

This "Edit site" link...

![CleanShot 2024-07-21 at 16 30 49](https://github.com/user-attachments/assets/565eef00-d93b-48c9-b9fd-889f8094f858)

...to always open here: 

![CleanShot 2024-07-21 at 16 35 54](https://github.com/user-attachments/assets/dd740be8-c598-4dfb-a9ce-c9c51ff7d550)

**Comment by richtabor** [comment]
This would also close https://github.com/WordPress/gutenberg/issues/63784, as the homepage is the page assigned to the top-level of the Site Editor (naturally).

**Comment by Mamaduka** [comment]
The proposal to revert #37850 in the #outreach channel might be worth sharing.

**Comment by ndiego** [comment]
In theory, I like the current implementation. I know how the Site Editor works through and through, so when I click on "Edit Site", I know what to expect. That said, if you are a newer user or not familiar with the Site Editor, it can be extremely disorienting. It immediately assumes you understand what templates are and how they are composed. So, I agree with the approach outlined above. It improves consistency as well as the new user experience.

As a side note, it would be cool if the Edit Page button opened the page/post/whatever you are viewing in the Site Editor with the correct template applied, but there is still a lot of Editor unification work needed before this is possible.

**Comment by bph** [comment]
@WordPress/outreach 
What do you think about this modification?

**Comment by markhowellsmead** [comment]
I agree that the current implementation could be confusing for many users: in fact, several of my clients continually misunderstand this link and think that they need to use it to edit the current page. (Particularly because “Edit site” and “Edit page” are translated almost identically in German.) When I work on sites, it's a huge advantage to use this link to jump directly into the relevant view within the Site Editor, but it might make the path for less-experienced users more easy to understand.

**Comment by bph** [comment]
I would like Edit Site with goes to the Site editor loading the homepage, if there were also a link to "edit template". 
As a site owner or theme builder, I would edit more often any the particular template rather the home page over and over. 

During the process of creating, I would navigate the site on the front end and then see something on that particular page, I would like to change. So I click on the "Edit template" button and come to the template. 

In that scenario, I would be quite annoyed if I had to get to the site editor and then Templates, scroll through the template and then edit it. 

So if it were to moved back to go to the Site editor > Homepage. 
The confusion comes from the disconnect strongly agree but as a theme builder, I would appreciate an Edit Template link off the top of the Admin bar.

**Comment by sethrubenstein** [comment]
I also agree that the current implementation can be confusing, specifically around naming. For that reason we modify the admin toolbar and rename this explicitly to "Edit Template", our users understood this better than "Edit Site" as its a more clear what the admin bar control does. 

That said, I would hate to lose existing functionality that we use day in and day out at this point. Making this  only open the site editor without any template context without something else in place to replace that lost functionality would not be ideal.

**Comment by bacoords** [comment]
I wrote about this 2 years ago and even wrote a small hacky plugin to get what I wanted. I think this UI is solved by other builders by making the 'Edit Site' button is a dropdown. Clicking it should go to the homepage of the site editor, but the dropdown gives you a quick reference to jump to the current template and any visible template parts, so it'd look like this: 

- Edit Site
  - Edit Header
  - Edit Archive
  - Edit Footer

I'd also be in favor of combining it with the 'Edit Post' link, the two "Edit" links is a weird experience.

Here's the original post: https://www.briancoords.com/writing-a-tiny-full-site-editing-plugin/

**Comment by andreawetzel** [comment]
> Most people just want to edit their site, and starting from one consistent point is a clear way to reduce the heaviness of navigating the site editing experience.

I agree with this. 

Also, I often accidentally click on "Edit site" when I intend to click on "Edit page" and so it takes me a few extra seconds to realize that I've landed on the template edit screen rather than the page edit screen. If I clicked on Edit site and landed on the top level of the site editor, I think it would relieve some mental load and offer a clear path to get to the page I wish to edit.

**Comment by richtabor** [comment]
> When I work on sites, it's a huge advantage to use this link to jump directly into the relevant view within the Site Editor, but it might make the path for less-experienced users more easy to understand.

> That said, I would hate to lose existing functionality that we use day in and day out at this point. Making this only open the site editor without any template context without something else in place to replace that lost functionality would not be ideal.

Do you use Create Block Theme? It could add a "Edit template" item to the toolbar, to get you directly to that template.

**Comment by richtabor** [comment]
> As a side note, it would be cool if the Edit Page button opened the page/post/whatever you are viewing in the Site Editor with the correct template applied, but there is still a lot of Editor unification work needed before this is possible

> I'd also be in favor of combining it with the 'Edit Post' link, the two "Edit" links is a weird experience.

Agreed, but not yet. 

This could be an interesting experiment (or plugin), but I do think we're a while off from a true 1:1 expectation wise between navigating the Site Editor and to/from posts and admin.

**Comment by colorful-tones** [comment]
I have zero expectations around orientation when I click 'Edit Site' or 'Go to Dashboard' (the top leftmost site icon in the Site Editor). I have not allowed myself the mindset to build _any muscle memory in any of these experiences over the years because of the changes like this_. I wouldn't say I like arguing for no change, but it worries me. 😢 

While it makes sense _now_ to change the Edit Site, just as much as it did to [do it back in 2022](https://github.com/WordPress/gutenberg/issues/37850), I worry that it'll be revisited in a year or two when the admin redesign gets further along. If so, then there is a risk of causing a lack of trust or perception of the quality of the project for everyday users. Risk and change is hard. I'd like to know how the priority of these particular changes sets in the bigger scheme of things and whether we're considering that. What is the urgency? Is there urgency? 🤔

**Comment by markhowellsmead** [comment]
I do, but I haven’t used it a great deal over the past few months because
of a different focus. I’ll take a closer look at the newer features for
sure.

**Comment by richtabor** [comment]
> I have not allowed myself the mindset to build any muscle memory in any of these experiences

Because it's been unpredictable—not because of a push for consistency. 

> I'd like to know how the priority of these particular changes sets in the bigger scheme of things and whether we're considering that. What is the urgency? Is there urgency?

I'm trying to make the WordPress experience more predicable and intuitive. The urgency is that there is an inconsistent experience that makes WordPress _feel_ hard. Sure, if one understands templates and knows that "Edit site" means to edit the current template, like "Edit page" does—then you choose it.

Otherwise, the thinking is that you can edit your site. You press it and you're instantly lost. You don't recognize the current view. Is it the editor? It looks like it, but there's no content, just a message saying that's where content will go (the template state of the post content block). 

These sort of seemingly tiny details add up quickly, making WordPress feel like the wrong choice. But if we push to meet expectations, it will feel like the right.

**Comment by luminuu** [comment]
I like the idea of @bacoords with submenu items, an alternative suggestion could be:

- Edit site
  - Navigation
  - Styles
  - Templates
  - Patterns

Kind of mimicking what items you have on the first page of the site editor. I've left out the Pages intentionally, as I think this should belong to a menu item specific for content like "Edit page"/"Edit post" is today.

**Comment by colorful-tones** [comment]
> Because it's been unpredictable—not because of a push for consistency.

It would be wonderful to hear more about the historical differences here and whether they align with the community's perspective or that of the individual.

> I'm trying to make the WordPress experience more predictable and intuitive. The urgency is that there is an inconsistent experience that makes WordPress feel hard. 

I worry that the "I" is emphasized, not the "we". So, the question of urgency should be proposed to the larger group and conversation around this small detail and request. Is there urgency, and will the impact of changing this small detail today outweigh the risk of needing a holistic overview of where the admin bar will exist in one year? Two years? 🤔 

It might be beneficial to remind folks of the WordPress [Philosophy](https://wordpress.org/about/etiquette/) and [Etiquette](https://wordpress.org/about/etiquette/), which is always good to revisit from time to time. 

It _could_ be beneficial to see some designs and discuss what and whether the Admin Bar will look like in 6 months to 1 year.

**Comment by ndiego** [comment]
> I worry that the "I" is emphasized, not the "we". So, the question of urgency should be proposed to the larger group and conversation around this small detail and request.

Rich is operating in good faith and is just trying to improve the editing experience in WordPress, just like we all are. The Admin Bar will undoubtedly see changes in the future, but I don't think we should be afraid to make incremental improvements to enhance the experience today, even though we know a more optimal "complete" solution may come down the line.

**Comment by richtabor** [comment]
> It could be beneficial to see some designs and discuss what and whether the Admin Bar will look like in 6 months to 1 year.

I'm not aware of plans to drastically change the admin bar. I want to take what we have today and remove friction, as is my typical approach to WordPress experience. 

> align with the community's perspective or that of the individual.

I would hope the community, like myself, wants the WordPress experience to be more intuitive. :) 

I have no motives other than making WordPress a little better every day.

**Comment by annezazu** [comment]
>Instead, I propose that this "Edit site" control should always open the Site Editor at the top level, just like the "Editor" control does within the WP Admin > Appearance menu—rather than on the template view of whichever page you're on.

I haven't seen this mentioned yet beyond sharing the issue number so I just want to bring some historical context to this thread in greater detail. Changing this would bring it back to the original implementation which was causing real confusion with users who expected that `Edit site` would bring them to what they were looking at. There's more feedback on the prior noted issue https://github.com/WordPress/gutenberg/issues/37850. This helped save folks from wading through the maze of templates to understand where they might want to edit something. Since it's landed, I haven't seen user feedback come in asking for a revert.

**Comment by bacoords** [comment]
The "Edit Site link as a dropdown" idea and variations of it have been mentioned by multiple people in both threads but hasn't received any recognition or response from any one on the editor team. Would love honest feedback on the idea and why it's not seen as a viable option.

References
- https://github.com/WordPress/gutenberg/issues/63785#issuecomment-2252763826
- https://github.com/WordPress/gutenberg/issues/63785#issuecomment-2250348987
- https://github.com/WordPress/gutenberg/issues/37850#issuecomment-1208422060
- https://github.com/WordPress/gutenberg/issues/37850#issuecomment-1221115551
- https://github.com/WordPress/gutenberg/issues/37850#issuecomment-1592642377
and so on.

**Comment by colorful-tones** [comment]
> Rich is operating in good faith and is just trying to improve the editing experience in WordPress, just like we all are. 

I do not believe I implied that he was not operating in good faith. However, explicitly calling this out shudders me to think that it is implied that I am not acting in good faith in trying to have a transparent and diplomatic dialogue with the project community. 🤔 

> just like we all are.

Yes, always assume positive intent. 🫶 

> I'm not aware of plans to drastically change the admin bar.

Neither am I. Thanks for clarifying.

> I want to take what we have today and remove friction, as is my typical approach to WordPress experience.

Wonderful! But how critical are these changes in the larger scheme of things? How certain are we that this change will not confuse everyday users, which would elevate the risk of introducing these changes? How do we assess and assign risk as a community towards seemingly minor changes but with the potential to have a huge impact (on 43% of the web) today, and tomorrow? 

> I have no motives other than making WordPress a little better every day.

I'm not questioning motives. I want to know how critical or urgent this overall change is in relation to many other crucial items.

Things like making WordPress "better" or "intuitive" are subjective. Ultimately, I'm grateful we're all here aiming for the same general goals.

I would assess this as medium (to high) risk with the potential to impact end user's trust in the user experience. With the 'Edit Site' link currently pointing to the current template in the Site Editor since 2022 - we risk confusing user expectations by introducing a change to point them to the Site Editor root. It may be more intuitive, but the risk outweighs the value. Overall, it would be ideal to focus on continuing to stabilize the Site Editor (a la Data Views). It would be wonderful to see us rally around #63128, #59745, #62371, #63255 

> The "Edit Site link as a dropdown" idea and variations of it have been mentioned by multiple people

This idea does seem to have some momentum and may be a potential solution.

**Comment by richtabor** [comment]
> Since it's landed, I haven't seen user feedback come in asking for a revert.

I don't think we'll see users asking to change the "Edit site" button. We'll see complaints about how navigating to/from/within the Site Editor is confusing and difficult.

Think if the "Customize" button in the Admin Bar opened in the different views depending on the page you were on. Not just different views, but different views that didn't look like your pages—wireframes (confusing wireframes) of your pages. That's what we have today with the "Edit site" admin link, at least my take.

**Comment by richtabor** [comment]
> The "Edit Site link as a dropdown" idea and variations of it have been mentioned by multiple people in both threads but hasn't received any recognition or response from any one on the editor team. Would love honest feedback on the idea and why it's not seen as a viable option.

@bacoords I only saw @luminuu's comment above a bit ago today and am thinking on it.  It's viable if we do what I'm suggesting here, otherwise likely more confusing to have an "Edit site" button that takes you multiple places, with a dropdown that also takes you multiple places.

**Comment by richtabor** [comment]
As-if, if folks _do not_ want to make this a direct link to the Site Editor (similar to the previous "Customize" link on classic themes), it should be renamed to "Edit template"—to better articulate what it's for.

Although, it's then touching on the confusion of template vs. content — especially when some templates _are_ content.

**Comment by bacoords** [comment]
> It's viable if we do what I'm suggesting here, otherwise likely more confusing to have an "Edit site" button that takes you multiple places, with a dropdown that also takes you multiple places.

Yeah there's more conversation needed, but I think my personal preference would be something like:

- Site Editor (the homepage of the site editor, original functionality)
  - Edit Header
  - Edit {Template} (dynamic to whatever template you're on, the current functionality)
  - Edit Footer
 
The two separate "Edit..." links in the admin bar feel "wrong", but by changing it to "Site Editor" it's similar to the way we used to have "Customize" to send you to the Customizer. That said I can see that I'm changing the language used from an action/verb to a destination/noun.

**Comment by richtabor** [comment]
> Site Editor (the homepage of the site editor, original functionality)
> Edit Header
> Edit {Template} (dynamic to whatever template you're on, the current functionality)
> Edit Footer

> The two separate "Edit..." links in the admin bar feel "wrong", but by changing it to "Site Editor" it's similar to the way we used to have "Customize" to send you to the Customizer. That said I can see that I'm changing the language used from an action/verb to a destination/noun.

Agreed. I think there's something here.

**Comment by carolinan** [comment]
I am one of the people who pushed for the link to point to the current template, and I do use it all the time, like literally every few minutes while I am working on themes or testing PR's.
But I can second that I have had conversations with users -not theme designers and developers - who found this confusing, and having needed to explain why the link opens to different places, and how to get the sidebar to show again (the menu with the dark background).

I would prefer the dropdown idea, with a small exception. I don't think the term "Site Editor" should be the link text. Because one of the long term goals is to have one editor.
Should the top level link be the same as the Appearance > Editor?

**Comment by carolinan** [comment]
There is an option to use the WP_DEVELOPMENT_MODE constant to determine where the links should go:
But since every user will have different needs, at that point, the developer may just as well add a must use plugin that adds their personal links to the bar.

Edit: Or maybe it should be under user preferences? I am only half joking.

**Comment by paaljoachim** [comment]
Brain storming....

Navigating through a site while logged in I am used to the experience of having useful items in the top black admin toolbar giving me the opportunity to do various useful things. The W top left I never use and really do not know why it is there. Seeing the site title drop with Dashboard, Plugins Themes can also contain a link to the top level area of the Editor. 

**Edit site link**. Which creates the expectation that this is similar to Edit Page / Edit Post that it would naturally go to the template being used for the page/post I was currently on. This was fixed and I am very happy for that. If current page is a post then clicking Edit site the link goes to the Single Posts template. If it is the front page then clicking the Edit site link goes to the Front pages template if one is used. If on a Shop WooCommerce page clicking Edit site link goes to the Product Catalog template.

The Edit site link has become very similar to the Edit Page / Edit Post link and that makes me very happy.

**Comment by JosVelasco** [comment]
I would love a description before clicking the Edit Site link. When hovering the element, a text could say, "Customize the appearance of this template using the block editor."

This could extend to other links. Adding more descriptions to other elements could help people understand old concepts, like the difference between a post and a page, even before clicking "New and selecting page or post."

**Comment by unscripted** [comment]
My two cents: We've had a few clients and non-dev team members get the "Edit site" and "Edit page/post" buttons confused. There are definitely merits to the current, proposed (old) and dropdown methods.

I'm heavily editing styles and patterns in the early part of a project. It would be awesome if the button opened the `site-editor.php` instead of the Blog Home template.

Once the initial styles are established, my focus shifts to working on templates, and I very rarely need to return to the styles editor. In this scenario, we've heard feedback that "edit site" and "edit page/post" aren't differentiated enough. 

I love the idea of the dropdown @bacoords shared, as it works for all project phases: setup, build and maintain.

If we don't implement a dropdown, I think changing the button label to "Edit template" becomes more descriptive and accurate.

If neither of these changes is made, I recommend keeping the templates as is. Our team ultimately spends more time on templates once the initial style setup is complete. 

We also desperately try to keep our clients out of the style editor, so nothing to make that easier. 😅

**Comment by jorgefilipecosta** [comment]
> > Site Editor (the homepage of the site editor, original functionality)
> > Edit Header
> > Edit {Template} (dynamic to whatever template you're on, the current functionality)
> > Edit Footer
> 
> > The two separate "Edit..." links in the admin bar feel "wrong", but by changing it to "Site Editor" it's similar to the way we used to have "Customize" to send you to the Customizer. That said I can see that I'm changing the language used from an action/verb to a destination/noun.
> 
> Agreed. I think there's something here.

Hi @richtabor, to clarify do you think we should explore this dropdown menu option instead of not having a dropdown and the button simply going to the homepage of the site editor? Or as a first step, we have a single button and we just make it link to the homepage of the site editor?

**Comment by richtabor** [comment]
> I am one of the people who pushed for the link to point to the current template, and I do use it all the time, like literally every few minutes while I am working on themes or testing PR's.

What's clear though is that folks not building themes or testing PRs are struggling to grasp what this is. Instead of editing their site (which is easily confused with "editing what I'm looking at"), people land in the template view of their current page, with what looks like a broken version of their site (post content block vs. page contents). 

> Hi @richtabor, to clarify do you think we should explore this dropdown menu option instead of not having a dropdown and the button simply going to the homepage of the site editor? Or as a first step, we have a single button and we just make it link to the homepage of the site editor?

Single button for sure, perhaps the dropdown like as proposed above, though how would the admin bar know about the header and footer template parts on the page? (copy to be determined).

![Image](https://github.com/user-attachments/assets/87aa4fb2-20df-4cf5-8d14-280144445aa3)

**Comment by carolinan** [comment]
> What's clear though is that folks not building themes or testing PRs are struggling

Yes I agree, that is what I was trying to convey, had you quoted the entire paragraph.

There is also an issue and pull request already opened for this on Trac: https://core.trac.wordpress.org/ticket/62368
which could be used as the "backport".

**Comment by ptesei** [comment]
Thank you @carolinan for bringing light to my issue!

I propose renaming the label to avoid confusion, as we tend to click on the first "edit" we encounter. Additionally, I suggest displaying it consistently across all pages, including admin and multisite admin. (I've tested this on a multisite setup, and the rules implemented by #62501 aren't functioning as intended. btw)

We're living in this strange period where two UIs coexist... I think it's okay to gradually introduce changes.

Reading all your comments has given me an epiphany. Since the data view is already called "Design"... why not change the label to "Design"? It's such a beautiful and universal word!

  ![Image](https://github.com/user-attachments/assets/bc75ccd1-7166-4bab-97dd-ddc3b5ce9633)
![Image](https://github.com/user-attachments/assets/2e18d147-52d5-4d31-8fd6-0f6ca1d81f73)

**Comment by ptesei** [comment]
according to github documentation the other common name is LICENSE all caps no extention. 
https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-license-to-a-repository
maybe can be interesting to add also this case

**Comment by richtabor** [comment]
If you have a static page set as your homepage via reading settings, this is the experience you get when you go to edit your site. 

It's pretty bad.  

https://github.com/user-attachments/assets/daa9d537-b0f1-462a-8b89-bd25a9322a5a

**Comment by richtabor** [comment]
> I propose renaming the label to avoid confusion, as we tend to click on the first "edit" we encounter. 

The name is not the issue, it's the expectation of what the label conveys and where you actually end up in the experience (though I do like the word _Design_" :) 

Can we get momentum on this? Best to do it early in the release cycle as well. 

At least start with:

- First, update "Edit site" to land at the top-level `wp-admin/site-editor.php`, always. 
- Second, investigate a template edit control, potentially in create block theme (_maybe_ core).

**Comment by ptesei** [comment]
@richtabor for me, the name is an issue (I keep clicking on it even though I perfectly know where it goes ^^, especially when I'm editing in French — it's a wider, longer blob of letters, exactly as @markhowellsmead was saying).
But, for the sake of this issue, let's stay on topic. I've updated the pull request, hopefully it will move forward.

**Comment by t-hamano** [comment]
#69271 has been submitted to change this link to be top-level instead of template. Many ideas were suggested here, and I would be grateful for feedback from everyone involved in this discussion on whether this PR can be shipped.

Note that the equivalent core patch to this PR has already been approved: https://github.com/WordPress/wordpress-develop/pull/8372

cc @audrasjb

**Comment by carolinan** [comment]
The issue is kept open so that the discussion about adding a dropdown with more options can continue.

**Comment by t-hamano** [comment]
Do we move this issue to the "Punted to 6.9" column? It looks like it will take some time to resolve all the issues.

**Comment by Mamaduka** [comment]
Yes, the remaining enhancements should be punted to WP 6.9.

**Comment by TheJeffr0** [comment]
Hello there folks! In WordPress 6.7.2, when I click the Edit Site link, everything I want to edit is displayed, including the Nav menu. But in WordPress 6.8 Beta 2, when I click the Edit Site link, the Nav Menu disappears. It looks like a bug. However, I now need to find the Show Template option and select it. This is a more cumbersome way of editing a site versus the experience in 6.7.2. 

First, how are users supposed to know that they need to configure Show Template to be enabled to edit their entire site. How did we go from "It just works" to "Now there's a small barrier to get over" with the Edit Site link? If I click on Edit Site, I expect to be able to edit everything I see on the site.

**Comment by Mamaduka** [comment]
> First, how are users supposed to know that they need to configure Show Template to be enabled to edit their entire site.

There's no need to configure this. The pages will display with "Show Templates" enabled for block themes by default unless you toggle that option.

P.S. There was a small bug in beta 2, which is now resolved - https://github.com/WordPress/wordpress-develop/pull/8499.

**Comment by TheJeffr0** [comment]
"_The pages will display with "Show Templates" enabled for block themes by default unless you toggle that option._"

We ran into an issue with this today in WP 6.8 Beta 3. If you create a new blank page with Show Template enabled and the theme template does not include a title block, there is no obvious way to edit the title without having to find that ‘show template’ toggle or insert a title block.

So, users can end up in a situation where they insert a Pattern on a Page without a way to give it a title, and it shows up as No Title in the backend.

**Comment by carolinan** [comment]
@TheJeffr0 that is not related to the Edit Site link.

May I remind you that there is a call for testing the WP 6.8 features, that is suitable for adding feedback:
https://make.wordpress.org/test/2025/03/04/help-test-wordpress-6-8/

**Comment by Mamaduka** [comment]
> If you create a new blank page with Show Template enabled and the theme template does not include a title block, there is no obvious way to edit the title without having to find that ‘show template’ toggle or insert a title block.

This seems like a very theme-specific issue. A theme can opt out of the default rendering mode for pages, or users could use the "Rename" action from the Document Settings panel.

## Screenshot

![Image](https://github.com/user-attachments/assets/8d6957e9-4667-41a2-9f7f-3d74922dc869)

**Comment by westonruter** [comment]
I will note that clicking "Customize" to open the Customizer from the frontend will take you to that URL in the Customizer preview. So "Edit Site" is (was) consistent with "Customize" in that way.

**Comment by marktimemedia** [comment]
> This seems like a very theme-specific issue.

I have strong thoughts in response to this, but also want to respect @carolinan's point that @TheJeffr0's post you are replying to was off-topic for this specific issue.

Since I'm here... from a UX perspective using FSE every day to build client sites, I am personally in favor of a distinction between "Edit Site" (go to main editor interface), "Edit Template" or "Edit {Template Name}" (go directly to editor for the template used on this page), and "Edit {Template Part Name}" (go directly to editor for header, footer, or other template parts present on the page) and the ability to navigate to those from the main admin bar, perhaps as a dropdown as suggested.

**Comment by Graphnic** [comment]
I reported this change as a bug yesterday without realising this was an intentional change.  While I can see the rationale for it, it also makes building sites with FSE immensely frustrating - what was once a single click to edit a template is now three clicks. 

@marktimemedia's suggestion of an 'Edit Template' seems to make a lot sense.  I knocked together this morning a very rough proof-of-concept: https://github.com/Graphnic/humaniti-edit-template-toolbar. It keeps the functionality separate from the 'Edit Site' button, while actually extending its usefullness.

**Comment by rmorse** [comment]
Agree, I think it makes building with FSE more difficult only having an `Edit Site` option and no easy way to edit the current template.

I'd rather keep the `Edit Template` functionality instead of `Edit Site` for the admin bar, with `Edit Site` being made available via the menu in wp-admin - I feel that contextual links are more important & convenient when viewing the frontend.

I really like the concept @Graphnic  - making it easy to choose which template to edit as there may be more than one currently loaded.

I think a variation on that, that could work is to keep the `Edit Site`  link, and have the various `Edit Template` links in a dropdown from `Edit Site`  - condensing the number of options in the admin bar related to site editing.

**Comment by Graphnic** [comment]
Yes, dropping the options down from Edit Site is probably neater, @rmorse.  As you say, the main thing is to preserve in some way the contextual nature of the feature that has been lost in 6.8.

**Comment by eduwass** [comment]
I'm here after 6.8 changed the previous functionality. In my opinion, a dropdown like what @Graphnic is proposing would be beneficial. Navigating through multiple menus to access what previously required just one click has disrupted the workflow in my scenario.

**Comment by t-hamano** [comment]
Thanks everyone for your feedback, I'll add this issue to the project board for discussion on whether something should be addressed in a minor release (6.8.x).

**Comment by t-hamano** [comment]
>While I can see the rationale for it, it also makes building sites with FSE immensely frustrating - what was once a single click to edit a template is now three clicks.

> Agree, I think it makes building with FSE more difficult only having an `Edit Site` option and no easy way to edit the current template.

> Navigating through multiple menus to access what previously required just one click has disrupted the workflow in my scenario.

It seems like a lot of users find this hard to use or that it's a bug. Furthermore, I've seen multiple similar reports.

On the other hand, the suggestion to add submenus seems to be generally well received.

What do you think about reverting [GB #69271](https://github.com/WordPress/gutenberg/pull/69271) and [Changeset 59910](https://core.trac.wordpress.org/changeset/59910) in the 6.8.x minor release, and then considering implementing a dropdown after that?

cc @Mamaduka @aaronjorbin @carolinan

**Comment by carolinan** [comment]
Only if you are estimating that the dropdown can not be included in one of the first minor releases.

**Comment by carolinan** [comment]
I mean it would not be ideal to change it 3 times within a span of say, 3 months.

**Comment by t-hamano** [comment]
Here's what I'm thinking:

- 6.8.1 (or 6.8.2?) Restore the Edit Site link to its previous link (link to templates)
- 6.9: Implement and ship the link with submenus

**Comment by SporeProductions** [comment]
Just add a new link/button that says "Edit Template" and have it appear when viewing a Page or Post, just like how "Edit Post" or "Customizer" only appear based on what type is being viewed. Problem solved and both UX paths are addressed. 

*One caveat may be the impact on non-Block themes?

**Comment by marktimemedia** [comment]
Wanted to chime in and say that an "Edit Template" functionality would have additional contexts beyond just a certain post type (several kinds of post or taxonomy archive templates, the "blog home" template, author templates, custom page templates, etc.) Having a way to directly jump to editing the template of whatever view you are currently looking at would bring this to UX parity with items like "Edit Post" (and also provide a bit of education of "oh, this is the template connected to this content/view, good to know")

**Comment by Graphnic** [comment]
Yes, exactly this @marktimemedia.  It could be very helpful to guide folks around the anatomy of a block theme.  It's not at all clear in the UI at the moment.

My very-quickly-hacked-together plugin ([linked earlier](https://github.com/WordPress/gutenberg/issues/63785#issuecomment-2809268325) in the thread) does this, and uses pre-existing WP functions to extract the available templates, and the template currently in use on the current view. There may well be edge-cases I missed, but it seems to work on the sites I've tested it on so far.

**Comment by carolinan** [comment]
Frankly, we don't need another plugin. What's needed is:

1. An agreement
2. A developer submitting the PR to the Gutenberg GitHub repository
3. A developer who does a code review of said PR
4. A tester who tests the functionality
5. A trac ticket for the core change
6. A PR for the core change
7. A review of the core change
8.  A core commit.

Is this statement harsh? Maybe, but I am seeing a lot of conversations, and very little code changes actually being submitted.
"We need a drop down with this and this link". Yes, and what gets us there is someone creating it.

**Comment by Graphnic** [comment]
Understood @carolinan. 

I don't know how an agreement is made here, or who is empowered to make an agreement - I assume the change in 6.8 was strategically discussed and agreed.

I'm not suggesting (another) plugin. Quite the opposite in fact. I only created it as a stop-gap for my personal use because the change in 6.8 was slowing my workflow considerably. Shared because I thought it might be helpful to someone and/or demonstrate the functionality being discussed. Perhaps the wrong place to do so. Apologies, I'll leave you all to it.

**Comment by westonruter** [comment]
> I'm not suggesting (another) plugin. Quite the opposite in fact. I only created it as a stop-gap for my personal use because the change in 6.8 was slowing my workflow considerably. Shared because I thought it might be helpful to someone and/or demonstrate the functionality being discussed. Perhaps the wrong place to do so. Apologies, I'll leave you all to it.

@Graphnic I think it's quite helpful to throw together POC plugins like you have done to demonstrate an approach. It can serve as the basis for a PR, and like you say it is a stop-gap measure while waiting for the PR to land in a release. I do this often myself. Thank you for doing it.

The question for others involved in this thread is whether your POC addresses the various scenarios. If so, it can move forward into a PR.

**Comment by carolinan** [comment]
I am not directing my comment to a single person. I have closed a few tickets linking to this one, but also had countless of people approach me and ask me about it, telling me "OK, well I have already created a custom plugin for myself that fixes it".
Not one, not three, _countless._ (And of these people only one person shared the code)

And that does not solve the issue for all WordPress (block theme) users, and that is what my frustration stems from.

**Comment by carolinan** [comment]
> Here's what I'm thinking:
> 
> * 6.8.1 (or 6.8.2?) Restore the Edit Site link to its previous link (link to templates)
> * 6.9: Implement and ship the link with submenus

I support this as long as the progress does not halt (again).

**Comment by t-hamano** [comment]
> 6.8.1 (or 6.8.2?) Restore the Edit Site link to its previous link (link to templates)

Unless someone has strong objections, let's go ahead with this.

> 6.9: Implement and ship the link with submenus

There are probably many different scenarios to consider and preferences may vary as to what is ideal, so any thoughts or PoCs are welcome.

**Comment by joemcgill** [comment]
> Unless someone has strong objections, let's go ahead with this.

@t-hamano I don't have strong feelings about whether the original change should be reverted in 6.8 while a new approach is being worked on, but if that is the preference of the Editor team, can you (or someone) make sure that there is a ticket added Trac for handling the revert so it can be tracked as part of minor release milestones?

As an observation, this issue has now grown to the point that it is a bit unclear when this issue can be considered "done" and whether new follow-up issues or sub-issues from this one should be spun up instead.

**Comment by t-hamano** [comment]
> can you (or someone) make sure that there is a ticket added Trac for handling the revert so it can be tracked as part of minor release milestones?

Core ticket: https://core.trac.wordpress.org/ticket/63358

**Comment by aaronjorbin** [comment]
[Revert](https://core.trac.wordpress.org/changeset/60193) has landed in WordPress Core trunk and is planned to be included in 6.8.1. Thanks everyone!

**Comment by t-hamano** [comment]
I think all the issues in 6.8.x have been resolved, so let's move it to the `WordPress 6.8  Editor Tasks > Punted to 6.9` column.

**Comment by t-hamano** [comment]
This issue has been punted to 6.9. Let's move this issue to the 6.9 Editor Tasks board and see what's possible in 6.9.

**Comment by t-hamano** [comment]
This ticket was featured on today's 6.9 Bug Scrub.

This issue needs some design and development to move forward; otherwise, it will be punted to 7.0.

**Comment by westonruter** [comment]
Related:

- #47934 

This is basically the other end of the user journey, for what URL you are taken to when you exit the Site Editor. As with the Customizer, I think exiting the Site Editor should take you back to whatever URL you had been on when you clicked "Edit Site".

**Comment by ryanwelcher** [comment]
Reviewed in todays 6.9 Bug Scrub. As we're 1 week away from Beta 1, I am going to punt this. If I am mistaken, please let me know.
